tasks/main.yml

```diff
@@ -1,25 +1,134 @@
 # SPDX-License-Identifier: MIT
 ---
-- name: Set platform/version specific variables
-  include_tasks: tasks/set_vars.yml
-
-# Examples of some tasks:
 - name: Ensure required packages are installed
   package:
-    name: "{{ __template_packages }}"
+    name: "{{ __systemd_packages }}"
     state: present
 
-- name: Ensure required services are enabled and started
-  service:
+- name: Deploy unit files
+  copy:
+    src: "{{ item }}"
+    dest: "{{ __admin_units_dir }}/{{ item }}"
+    owner: root
+    group: root
+    mode: '0644'
+  when: systemd_unit_files is defined
```

Comment:
```suggestion
```
because `systemd_unit_files` is defined in defaults/main.yml, it will always be defined - you can just use the `loop` which will be a no-op if `system_unit_files` is `[]` (empty)

Reply:
Hmm, when I use `loop` instead of `with_items` and `systemd_unit_files` is `[]` I am getting this error,

`(item=[]) => {"ansible_loop_var": "item", "changed": false, "item": [], "msg": "src (or content) is required"}`

Reply:
I guess I need to check `length` of list and loop over it only if it is non-empty.

Reply:
You should use `loop` as `loop: "{{ systemd_unit_files }}"`, not as
```
loop:
  - "{{ systemd_unit_files }}"
```

Reply:
I will just stick to `with_items` for now.

Reply:
Ahh, I misread previous comment. Indeed that works. Sorry for the noise.